Вичурин Н.М. Лаб. 4 Группа 6311

VichurinNik opened the pull request:
**ФИО:** Вичурин Никита
**Номер группы:** 6311  
**Номер лабораторной:** 4
**Краткое описание исходного кода:** Я взял(а) исходный код  вот от сюда: https://github.com/positivepeng/nlp-beginner-projects/tree/master/project5-Text%20Generation
**Краткое описание проделанной работы:** 

1: Сделали всё аккуратнее в одном классе
Создали класс PoetryTrainer где всё лежит по полочкам

2: Все настройки в одном месте
Все настройки в одном словаре config

3: Добавил поиск ошибок
Код умеет обрабатывать ошибки и понятно объяснять что случилось

4:Прогресс-бар чтобы видеть как идет обучение
Авто-стоп когда модель перестает улучшаться
Сохранение только лучшей версии модели

Сделали удобный интерфейс
Можно просто запустить и вводить слова для генерации стихов

IlyaOv commented on lab_3/main.py:
```diff
@@ -0,0 +1,56 @@
+        "inn": r"\d{10}|\d{12}",
+        "identifier": r"\d{2}-\d{2}/\d{2}",
+        "ip_v4": r"\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}",
+        "latitude": r"-{0,1}\d{1,2}\.\d*",
```
Для широты стоит указать ограничения, которые она имеет.

IlyaOv commented on lab_3/main.py:
```diff
@@ -0,0 +1,56 @@
+        "blood_type": r"(A|B|AB|O)[+−]{1}",
+        "isbn": r"(\d{3}-){0,1}\d-\d{5}-\d{3}-\d",
+        "uuid": r"[a-f\d]{8}-([a-f\d]{4}-){3}[a-f\d]{12}",
+        "date": r"\d{4}-\d{2}-\d{2}",
```
Дату тоже предлагаю уточнить.

VichurinNik committed:
Заменил строковые константы на Enum

VichurinNik committed:
Разделенил большую функцию на маленькие

VichurinNik committed:
Создание класса-обработчика данных

VichurinNik committed:
Улучшил подготовку данных
Улучшил конвертацию
Улучшил создание DataLoader
Упростил интерфейс для быстрого использования
Добавил докстринги

VichurinNik committed:
Созданил класса тренера с конфигурацией

VichurinNik committed:
Улучшил инициализацию компонентов

VichurinNik committed:
Улучшен цикл обучения

VichurinNik committed:
Добавил русские докстринги и переделал main

VichurinNik committed:
con

IlyaOv reviewed (approved):
Зачтено
